Fedoraware/Fedoraware-TF2/src/Features/Vars.h

```diff
@@ -2,920 +2,928 @@
-		inline CVar<int> SeperatorSpacing{ 6 };
-		
-		inline CVar<int> ParticleTracer{ 2 };
-		inline std::string ParticleName = "merasmus_zap_beam01"; // dont save this as a var its pointless
-		inline CVar<bool> DoPostProcessing{ false };
-
-		namespace Beans
-		{
-			inline CVar<bool> Active{ false };
-			inline CVar<bool> Rainbow{ false };
-			inline Color_t BeamColour{ 255, 255, 255, 255 };
-			inline CVar<bool> UseCustomModel{ false };
-			inline std::string Model = "sprites/physbeam.vmt";
-			inline CVar<float> Life{ 2.f };
-			inline CVar<float> Width{ 2.f };
-			inline CVar<float> EndWidth{ 2.f };
-			inline CVar<float> FadeLength{ 10.f };
-			inline CVar<float> Amplitude{ 2.f };
-			inline CVar<float> Brightness{ 255.f };
-			inline CVar<float> Speed{ 0.2f };
-			inline CVar<int> Flags{ 65792 };
-			inline CVar<int> Segments{ 2 };
-		}
-
-		inline CVar<bool> BulletTracerRainbow{ false };
-		inline CVar<bool> AimPosSquare{ false };
-		inline CVar<bool> OutOfFOVArrows{ false };
-		inline CVar<float> ArrowLength{ 21.f };
-		inline CVar<float> ArrowAngle{ 100.f };
-		inline CVar<float> MaxDist{ 1000.f };
-		inline CVar<float> MinDist{ 200.f };
-		inline CVar<int> VMOffX{ 0 };
-		inline CVar<int> VMOffY{ 0 };
-		inline CVar<int> VMOffZ{ 0 };
-		inline Vec3 VMOffsets{};
-		inline CVar<int> VMRoll{ 0 };
-
-		inline CVar<float> NotificationLifetime{ 5.f };
-		inline CVar<bool> DamageLoggerText{ false };
-		inline CVar<bool> DamageLoggerChat{ false };
-		inline CVar<bool> DamageLoggerConsole{ false };
-
-		inline CVar<int> VisionModifier{ false };
-
-		inline CVar<int> Rain{ false };
-		inline CVar<bool> EquipRegionUnlock{ false };
-
-		inline CVar<bool> NoStaticPropFade{ false };
-		inline CVar<bool> ParticlesIgnoreZ{ false };
-
-
-		namespace RagdollEffects
-		{
-			inline CVar<bool> EnemyOnly{ false };
-			inline CVar<bool> Burning{ false };
-			inline CVar<bool> Electrocuted{ false };
-			inline CVar<bool> BecomeAsh{ false };
-			inline CVar<bool> Dissolve{ false };
-			inline CVar<int> RagdollType{ 0 };
-		}
-
-		namespace Skins
-		{
-			inline CVar<bool> Enabled{ false };
-			inline CVar<int> Sheen{ 0 };
-			inline CVar<int> Effect{ 0 };
-			inline CVar<int> Particle{ 0 };
-			inline CVar<bool> Acient{ false };
-			inline CVar<bool> Override{ false };
-			inline CVar<bool> Australium{ false };
-		}
-
-		namespace Fog
-		{
-			inline CVar<bool> DisableFog{ false };
-			inline CVar<bool> CustomFog{ false };
-			inline Color_t FogColor{ 255, 255, 255, 255 };
-			inline Color_t FogColorSkybox{ 255, 255, 255, 255 };
-			inline CVar<float> FogDensity{ 0 };
-			inline CVar<float> FogDensitySkybox{ 0 };
-			inline CVar<float> FogStart{ 0 };
-			inline CVar<float> FogStartSkybox{ 0 };
-			inline CVar<float> FogEnd{ 0 };
-			inline CVar<float> FogEndSkybox{ 0 };
-		}
-	}
+#define CVar(name, value, ...) \
+	inline ConfigVar<decltype(value)> name{ value, GetNamespace() + GetSubname() + std::string(#name), __VA_ARGS__ }; \
```

Comment:
Would probably better without the `;` at the end, so we're forced to add a semicolon after using the macro (also i think this messes with formatting, at least when using ReSharper)